### PR TITLE
Clarify request authentication root namespace scope.

### PR DESCRIPTION
**Pull request by incfly**
No behavior change just clarification.

We have the clear documentation in authorization policy of what would happen for "istio-system" authz, but not request authn.
Verified the behavior is as documented in this PR. This would give some clarify of the use case.

Also to give some pointers for the impl,
https://github.com/istio/istio/blob/37b717fc897730710e224c1f0e55303e374d22d9/pilot/pkg/model/authentication.go#L231.

### CODE REVIEW

**Comment by incfly** [comment]
where authz is doc, https://github.com/istio/api/blob/2af152115f81571984b4146e93274f82ecc73f3e/security/v1beta1/authorization_policy.proto#L137

File: security/v1beta1/request_authentication.proto [hunk 1]
```diff
@@ -64,6 +64,21 @@ option go_package="istio.io/api/security/v1beta1";
 //         requestPrincipals: ["*"]
 // ```
 //
+// - A policy in root namespace ("istio-system") applies to workloads in all namespaces in a mesh.
+// The following policy makes all sidecars perform JWT authentiation.
```
Comment (yangminzhu) [inline security/v1beta1/request_authentication.proto:68]: not only sidecars, I think this also applies to gateways.

Comment (incfly) [reply security/v1beta1/request_authentication.proto:68]: good point. updated.

**Comment by incfly** [comment]
friendly ping, thanks.

@smawson @howardjohn @nrjpoddar @brian-avery

File: security/v1beta1/request_authentication.proto [hunk 2]
```diff
@@ -64,6 +64,33 @@ option go_package="istio.io/api/security/v1beta1";
 //         requestPrincipals: ["*"]
 // ```
 //
+// - A policy in root namespace, "istio-system" by default, applies to workloads in all namespaces
```
Comment (smawson) [inline security/v1beta1/request_authentication.proto:67]:
```suggestion
// - A policy in the root namespace ("istio-system" by default) applies to workloads in all namespaces
```

File: security/v1beta1/request_authentication.proto [hunk 3]
```diff
@@ -64,6 +64,33 @@ option go_package="istio.io/api/security/v1beta1";
...
+// in a mesh. The following policy makes all workloads only accept the requests that contain a
```
Comment (smawson) [inline security/v1beta1/request_authentication.proto:68]:
```suggestion
// in a mesh. The following policy makes all workloads only accept requests that contain a
```

**Review by linsun — APPROVED** [review]
LGTM after @smawson's suggestions

**Review by smawson — APPROVED** [review]

**Pull request merged**